app/com/linkedin/drelephant/tuning/Flow.java

```diff
@@ -0,0 +1,71 @@
+package com.linkedin.drelephant.tuning;
+
+import com.linkedin.drelephant.tuning.Schduler.AzkabanJobStatusManager;
+import com.linkedin.drelephant.tuning.engine.MRExecutionEngine;
+import com.linkedin.drelephant.tuning.engine.SparkExecutionEngine;
+import com.linkedin.drelephant.tuning.hbt.AlgorithmManagerHBT;
+import com.linkedin.drelephant.tuning.hbt.BaselineManagerHBT;
+import com.linkedin.drelephant.tuning.hbt.FitnessManagerHBT;
+import com.linkedin.drelephant.tuning.obt.AlgorithmManagerOBT;
+import com.linkedin.drelephant.tuning.obt.BaselineManagerOBT;
+import com.linkedin.drelephant.tuning.obt.FitnessManagerOBT;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class Flow {
+  Map<String, List<Manager>> pipelines = null;
+
+  public Flow() {
+    pipelines = new HashMap<String, List<Manager>>();
```

Comment:
We can think of Flow as 4 steps, running in parallel.
1. Baseline
2. Job status from Scheduler.
3. Fitness Manager
4. Parameter generation

Instead of having everything in the class Flow, which is primarily a driver class, should we have one interface for Tuning type and have implementations for it. This implementation can then give us further implementations for that specific tuning type for baseline, fitness compute and parameter generation.
This interface for tuning type would then be point of reference whenever we add a new tuning type.
Similarly we can do for scheduler in JobStatus manager.